crml/eth-bridge/src/lib.rs

```diff
@@ -212,10 +222,21 @@ decl_module! {
 				ProcessedTxBuckets::remove_prefix(expired_bucket_index, None);
 
 				// TODO: better estimate
-				50_000_000_u64
-			} else {
-				Zero::zero()
+				weight += 50_000_000 as Weight;
 			}
+
+			if DelayedEventProofs::iter().count() == 0 {
```

Comment:
this will iterate everything to count, maybe try  `decode_len()` or `iter().next().is_none()`
```suggestion
			if DelayedEventProofs::iter().count() == 0 {
```